include/seqan3/io/views/async_input_buffer.hpp

```diff
@@ -0,0 +1,486 @@
+//  async_input_buffer_fn (adaptor definition
+// ============================================================================
+
+//!\brief Definition of the range adaptor object type for seqan3::views::async_input_buffer.
+struct async_input_buffer_fn
+{
+    //!\brief Store the argument and return a range adaptor closure object.
+    constexpr auto operator()(size_t const buffer_size) const
+    {
+        return detail::adaptor_from_functor{*this, buffer_size};
+    }
+
+    /*!\brief Directly return an instance of the view, initialised with the given parameters.
+     * \param[in] urange      The underlying range.
+     * \param[in] buffer_size The frame that should be used for translation.
+     * \returns A range of translated sequence(s).
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, size_t const buffer_size) const
+    {
+        static_assert(std::ranges::input_range<urng_t>,
+            "The range parameter to views::async_input_buffer must be at least an std::ranges::input_range.");
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::async_input_buffer cannot be a temporary of a non-view range.");
+        static_assert(std::movable<std::ranges::range_value_t<urng_t>>,
+            "The range parameter to views::async_input_buffer must have a value_type that is std::movable.");
+        static_assert(std::constructible_from<std::ranges::range_value_t<urng_t>,
+                                              std::remove_reference_t<std::ranges::range_reference_t<urng_t>> &&>,
+            "The range parameter to views::async_input_buffer must have a value_type that is constructible by a moved "
+            "value of its reference type.");
+
+        if (buffer_size == 0)
+            throw std::invalid_argument{"The buffer_size parameter to views::async_input_buffer must be > 0."};
+
+        return detail::async_input_buffer_view{std::forward<urng_t>(urange), buffer_size};
+    }
+};
+
+}  // seqan3::detail
+
+//-----------------------------------------------------------------------------
+// View shortcut for functor.
+//-----------------------------------------------------------------------------
+
+namespace seqan3::views
+{
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief A view adapter that returns a concurrent-queue-like view over the underlying range.
+ * \tparam urng_t         The type of the range being processed. See below for requirements.
+ * \param[in,out] urange  The range being processed.
+ * \param[in] buffer_size Size of the buffer. Choose the size (> 0) depending on the expected work per element.
+ * \returns A view that pre-fetches elements from the underlying range and provides a thread-safe interface.
+ *          See below for the properties of the returned range.
+ * \ingroup views
+ *
+ * \details
+ *
+ * \header_file{seqan3/range/views/async_input_buffer.hpp}
```

Comment:
This line also needs adjustment:
```suggestion
 * \header_file{seqan3/io/views/async_input_buffer.hpp}
```